### ISSUE TITLE
Testing VotingPeriod update

### ISSUE THREAD

**Issue by ValarDragon** [osmosis-labs/osmosis#1138]
An idea thats been floating with lots of support on commonwealth is raising the voting period from 3 days to 5 days. Unfortunately the blocker for this going on-chain is testing that a voting length increase breaks nothing on testnet. I view what we need to test to ensure this works as follows:

For simplicity, lets say that voting period I start with is 5 minutes, and then I go to 20 minutes. (Can do whatever times) 

At voting period of time T, pre-upgrade:
(1) Have a proposal thats already finished voting > 20 minutes ago.
(2) Have a proposal thats already finished voting, but under 15 minutes ago.
(3) Have a proposal thats in voting
(4) have a proposal thats in deposit phase.

Do parameter change proposal to 20 minutes. (2) at time of param change should be through voting, but not 15 minutes through voting.

Check that:
- Prop 1's end voting time did not change
- Prop 2's end voting time did not change / its decision is final. Check again on this once its 20 min passed prop creation
- See what happens with prop 3, does voting end time change or no? Does it end when voting end time says it'll end / is there any oddities
- Move prop 4 to voting, see what its end time is. Double check that it ends when it says it should
- Try making a new prop (5), and just double check it has 20 minutes as voting period, and it ends in 20 minutes

**Comment by alexanderbez** [comment]
Is the issue or concern here that increasing the voting period slightly would impact proposals that in the current voting period?

**Comment by czarcas7ic** [comment]
**TLDR: All checks out, we should be good to make this param change in prod.**

****

I tested the following using the local multinode script. The only thing I started off differently with was initial voting time to 300 seconds (5 minutes). @ValarDragon If you are happy with the following testing and have no further edge cases, you can close the issue and I will kill the VM.

## Commands / Sequence

Prop 1: Set max validators to 110
```
osmosisd tx gov submit-proposal param-change proposal1.json --from validator1 --keyring-backend=test --home=$HOME/.osmosisd/validator1 --chain-id=testing -b block --yes
osmosisd tx gov deposit 1 10000000uosmo --from=validator1 --keyring-backend=test --chain-id=testing --broadcast-mode=block --yes --home=$HOME/.osmosisd/validator1
osmosisd tx gov vote 1 yes --from=validator1 --keyring-backend=test --chain-id=testing -b block --yes --home=$HOME/.osmosisd/validator1
osmosisd tx gov vote 1 yes --from=validator2 --keyring-backend=test --chain-id=testing -b block --yes --home=$HOME/.osmosisd/validator2
```

Waited 20 minutes. Prop 2: Set max validators to 120.
```
osmosisd tx gov submit-proposal param-change proposal2.json --from validator1 --keyring-backend=test --home=$HOME/.osmosisd/validator1 --chain-id=testing -b block --yes
osmosisd tx gov deposit 2 10000000uosmo --from=validator1 --keyring-backend=test --chain-id=testing --broadcast-mode=block --yes --home=$HOME/.osmosisd/validator1
osmosisd tx gov vote 2 yes --from=validator1 --keyring-backend=test --chain-id=testing -b block --yes --home=$HOME/.osmosisd/validator1
osmosisd tx gov vote 2 yes --from=validator2 --keyring-backend=test --chain-id=testing -b block --yes --home=$HOME/.osmosisd/validator2
```

Waited for 5 minutes to pass. Initialize but do not start Prop 3: Set max validators to 130.
```
osmosisd tx gov submit-proposal param-change proposal3.json --from validator1 --keyring-backend=test --home=$HOME/.osmosisd/validator1 --chain-id=testing -b block --yes
```

Leave Prop 4: Set max validators to 140 in deposit period
```
osmosisd tx gov submit-proposal param-change proposal4.json --from validator1 --keyring-backend=test --home=$HOME/.osmosisd/validator1 --chain-id=testing -b block --yes
```

Prop 5: Set voting time from 300s to 1200s
```
osmosisd tx gov submit-proposal param-change proposal5.json --from validator1 --keyring-backend=test --home=$HOME/.osmosisd/validator1 --chain-id=testing -b block --yes
osmosisd tx gov deposit 5 10000000uosmo --from=validator1 --keyring-backend=test --chain-id=testing --broadcast-mode=block --yes --home=$HOME/.osmosisd/validator1
osmosisd tx gov vote 5 yes --from=validator1 --keyring-backend=test --chain-id=testing -b block --yes --home=$HOME/.osmosisd/validator1
osmosisd tx gov vote 5 yes --from=validator2 --keyring-backend=test --chain-id=testing -b block --yes --home=$HOME/.osmosisd/validator2
```

Wait 2 minutes, then get Prop 3 voting to start.
```
osmosisd tx gov deposit 3 10000000uosmo --from=validator1 --keyring-backend=test --chain-id=testing --broadcast-mode=block --yes --home=$HOME/.osmosisd/validator1
osmosisd tx gov vote 3 yes --from=validator1 --keyring-backend=test --chain-id=testing -b block --yes --home=$HOME/.osmosisd/validator1
osmosisd tx gov vote 3 yes --from=validator2 --keyring-backend=test --chain-id=testing -b block --yes --home=$HOME/.osmosisd/validator2
```

## Checks

### Prop 1's end voting time did not change
Did not change.

### Prop 2's end voting time did not change / its decision is final. 
Did not change. The only thing I sadly realized after all the tests was the following props overrided this prop so the 20 minute check wasn't checked. I can re-run with a separate param change if this is something we want to ensure.

### See what happens with prop 3, does voting end time change or no?
Voting time starts off at 5 minutes. Once prop 5 passes, the end time stays at 5 minutes. When 5 minutes pass, prop does in fact execute. 

### Move Prop 4 to voting. See what its end time is. Double check that it ends when it says it should
End time is 20 minutes. When greater than 5 minutes passes, checked param, did not change. When 20 minutes goes by, checked param, changed as expected.

### Make Prop 6, just double check it has 20 minutes as voting period, and it ends in 20 minutes
Has 20 minute voting period and behaves as expected.

## Files

proposal1.json
```
{
  "title": "Staking Param Change",
  "description": "Update max validators",
  "changes": [
    {
      "subspace": "staking",
      "key": "MaxValidators",
      "value": 110
    }
  ]
}
```

proposal2.json
```
{
  "title": "Staking Param Change",
  "description": "Update max validators",
  "changes": [
    {
      "subspace": "staking",
      "key": "MaxValidators",
      "value": 120
    }
  ]
}
```

proposal3.json
```
{
  "title": "Staking Param Change",
  "description": "Update max validators",
  "changes": [
    {
      "subspace": "staking",
      "key": "MaxValidators",
      "value": 130
    }
  ]
}
```

proposal4.json
```
{
  "title": "Staking Param Change",
  "description": "Update max validators",
  "changes": [
    {
      "subspace": "staking",
      "key": "MaxValidators",
      "value": 140
    }
  ]
}
```

proposal5.json
```
{
  "title": "Gov Param Change",
  "description": "Update voting period",
  "changes": [
    {
      "subspace": "gov",
      "key": "votingparams",
      "value": { "voting_period": "1200000000000" }
    }
  ]
}
```

proposal6.json
```
{
  "title": "Staking Param Change",
  "description": "Update max validators",
  "changes": [
    {
      "subspace": "staking",
      "key": "MaxValidators",
      "value": 160
    }
  ]
}
```

**Comment by ValarDragon** [comment]
> Is the issue or concern here that increasing the voting period slightly would impact proposals that in the current voting period?

Yeah that was the exact concern. Since all parameter validation in the SDK is stateless atm, I want to be extra-careful around us ensuring on a testnet that the param change won't break things! (We already bricked govenrance once 😅 )

**Comment by ValarDragon** [comment]
Awesome! Thank you so much for testing this adam!
